### PR TITLE
feat: Rename theme to context

### PR DESCRIPTION
**Pull request by castastrophe**
tldr; Updating references to theme to reflect the change in nomenclature to "context".

- Added documentation updates clarifying a few nuanced concepts around theming and context to ensure the distinction between the two concepts is clear.

### Preview

Link(s) to demo page(s) where this element can be viewed:
- [Link](https://deploy-preview-996--happy-galileo-ea79c4.netlify.app/)

### Testing instructions

1. Regression testing should be all that is needed; only minor variable and mixin name changes were made.

#### Browser requirements

Sanity check IE:
- [x] Internet Explorer 11 (should be useable, not pixel perfect)

### Ready-for-merge Checklist

Check off items as they are completed.  Feel free to delete items if they are not applicable.

- [x] Expected files: all files in this pull request are related to one request or issue (no stragglers or scope-creep).
- [x] Browser testing passed.
- [x] Repository compiles and tests pass.
- [x] Changelog updated (not needed for documentation updates).
- [x] Documentation (README.md, WHY.md, etc.) updated or added.

### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**

### CODE REVIEW

File: docs/assets/pfe-base-sass/functions/_functions.scss [hunk 1]
```diff
@@ -38,7 +38,7 @@
 }
 
 // New color function to only return theme colors
-@function color( $name, $theme: "light", $opacity: 1 ) {
+@function color( $name, $context: "light", $opacity: 1 ) {
```
Comment (castastrophe) [inline docs/assets/pfe-base-sass/functions/_functions.scss:41]: Todo: there's a few missed updates in this file

File: docs/content/theme/pfe-theme-slots.md [hunk 2]
```diff
@@ -97,66 +93,66 @@ At first glace, there seem to be lots of "gotchas" related to web components, an
...
-
+    <pfe-card>
+      <p slot="pfe-card--footer">World</p>
+      <h3 slot="pfe-card--header">Hello</h3>
+    </pfe-card>
 
 ## Styling Slots
 
-The lines blur between shadow DOM & light DOM when slots are involved. Basically if you add the attribute `slot=" "` to a regular HTML element inside a web component, you are opening a window to allow styles from the web component to style that thing. It only applies directly to the item with the slot name on it though, nothing nested inside it.
+If you add a `slot` to a light DOM element, you are opening a window to allow styles from outside the web component to style that section.  That means the typical encapsulation of a web component flies out the window. You can influence the styles using the `::slotted` selector but it only applies to the direct children of the slot.
```
Comment (starryeyez024) [inline docs/content/theme/pfe-theme-slots.md:114]: I'm confused who the "you" is in this sentence. Is it the developer building the component or the dev building the page with a component? Given the documentation page, I'm guessing its a dev building a component.

Can you "add slots to light DOM elements"? Or do you add slots to the shadow DOM, which allows content to be placed in the light DOM that then appears in the template of the component?

```suggestion
While building a component, if a `slot` is added, it opens a window to allow styles from outside of the component to style that section.  Slotted content foregoes the benefits of encapsulated styles. The component stylesheet may influence the styles using the `::slotted` selector, but it only applies to the direct children of the slot.
```

File: docs/content/theme/theming-for-older-browsers.md [hunk 3]
```diff
@@ -9,40 +9,33 @@ menu = "theme"
...
+
+    You can override either of the two provided variables, `--ie-fake-var` or `--another`.
+
+* IE11 will ignore slotted styles, it doesnt matter if you style it with `::slotted()`, `slot::slotted()` or even if you add a class to the children with JavaScript & try to style that. IE11 ignores all of it because of how it is restructuring the DOM using ShadyDOM.
 
-    and you can override either of those variables!
+## Safari
 
-2. IE11 will ignore slotted styles, it doesnt matter if you style it with `::slotted()` `slot::slotted()` or even if you add a class to the children with JavaScript & try to style that. IE11 ignores all of it.
+@TODO: Add some notes here on nuances with Safari
```
Comment (starryeyez024) [inline docs/content/theme/theming-for-older-browsers.md:41]: flagging this in case you wanted to complete it before merging

**Review by starryeyez024 — CHANGES_REQUESTED** [review]
much awesomeness, context is so much clearer than theme!

File: docs/content/develop/_index.md [hunk 4]
```diff
@@ -20,16 +20,16 @@ Before getting started, it's important to note that because this is a library, e
...
     - If it is a `combo` component, then it should not contain any styles of its own (exception: you might need to set the display value of the host) but rather, it will pull together other components into a set that is logical for content editors or developers. These components can also make decisions about the way the children components are laid out or which components are called based on built-in logic for that combo.
 3. Components should be context aware, if possible.
     - `Content` components should be equipped with styles for light, dark, and saturated backgrounds.
 4. Framework agnostic
-    - A PatternFly Element should "just work" when you drop it onto any page (provided the proper polyfills are there). It should have ALL the styles it needs, when coupled with a PFE Theme.
+    - A PatternFly Element should "just work" when you drop it onto any page (provided the proper polyfills are there). It should have ALL the styles it needs, and can be optionally coupled with a theme.
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply docs/content/develop/_index.md:30]: I like the idea of it being a link but let's call that a TODO for now because it doesn't seem critical to getting this out the door.

File: docs/content/develop/step-2c.md [hunk 5]
```diff
@@ -119,12 +119,12 @@ $LOCAL-VARIABLES: (
 .pfe-cool-element {
   &__profile {
     background-color: pfe-local(BackgroundColor, $region: profile);
-    // Any time background color is updated, theme should be set too
+    // Any time background color is updated, context should be set too
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply docs/content/develop/step-2c.md:122]: That seems more confusing to me and not quite right.  *Any* time the background is changed, whatever you're changing it to doesn't matter, a context should be defined to match.

File: docs/content/theme/_index.md [hunk 6]
```diff
@@ -1,42 +1,68 @@
...
 menu = "theme"
 tags = [ "theme" ]
+wcs = [ "pfe-cta", "pfe-card" ]
 +++
 
-
 ## Theming 101
 
-Every PatternFly Element is built to automatically utilize the colors defined in the [palette](/theme/_3_pfe-color-palette/), which you control! Generally speaking, the only thing you will need to do is re-define the CSS variables to match your brand and you're done.
+When referring to the "theme" in PatternFly Elements, we are talking about a set of CSS variables that, when built into the styles of the components, allow for a consistent styling across a site. Theme values are generically named using a BEM approach and focus on broad design concepts which makes them reusable in a variety of contexts.
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply docs/content/theme/_index.md:15]: generically - meaning not specifically.  I think that's correct.  It's not that different from generally but is more precise to our meaning.

[the remark this answers is not in the document]
Comment (castastrophe) [reply docs/content/theme/_index.md:15]: Component-specific variables aren't really a part of theming though so I don't think we muddy the waters here.

File: docs/content/theme/_index.md [hunk 7]
```diff
@@ -1,42 +1,68 @@
...
-	    </pfe-cta>
-	</pfe-card>
+```css
+:root {
+	--pfe-theme--color--ui-accent: #e00;
+}
+```
 
-Should you need to deviate from this color usage, and set your primary CTA to use the complement color from the palette, you may also pass a value of `complement` into the `pfe-color`  attribute, like this:
+(You don't have to use `:root`, it's just a common example in documentation related to CSS variables.)
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply docs/content/theme/_index.md:48]: This is quite verbose.  I find keeping things straight forward is easier to grok.

How about:

```suggestion
You don't have to use the `:root` selector; you can scope variables so they target the whole document (via `:root` or `body`) or specific areas (via an id or class).  CSS variables follow all the same rules as the standard CSS cascade.
```

File: docs/content/theme/_index.md [hunk 8]
```diff
@@ -1,42 +1,68 @@
...
 
-	<pfe-cta priority="primary" color="complement">
-		<a href="#">Primary</a>
+Any other component making use of that same `ui-accent` value will inherit your customization creating a consistent experience for your users.
+
+## Context
+
+Each component comes equipped to adjust its colors depending on the context inside which it exists. For example, should you need to put a default CTA (which is blue) on a dark blue card, the CTA will need to adapt so that the text remain readable. You can do this by informing the component of its context by giving the `context` attribute the value of `dark`.
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply docs/content/theme/_index.md:54]: Fair enough but you're example takes it in a totally different direction which was not what I was trying to communicate here at all.  I'll rework below.

Comment (castastrophe) [inline docs/content/theme/_index.md:54]:
```suggestion
Each component comes equipped to adjust its colors depending on the context inside which it exists. For example, should you need to put a default CTA (which is blue) on a dark blue card, the CTA will need to adapt so that the text remain readable. The card component therefore needs to pass down a `context` to the CTA, which it does using the `on="dark"` attribute.  This is programmatically applied by JavaScript that runs from the base class, `PFElement` (you don't have to do a thing!).  To manually override a component's context, you can do this by setting the `context="dark"` attribute.
```

File: docs/content/theme/_index.md [hunk 9]
```diff
@@ -1,42 +1,68 @@
...
+
+<pfe-card pfe-color="darkest">
```
Comment (castastrophe) [inline docs/content/theme/_index.md:56]:
```suggestion
<pfe-card color="darkest">
```

File: docs/content/theme/theming-for-older-browsers.md (hunk 3 above)
Comment (castastrophe) [reply docs/content/theme/theming-for-older-browsers.md:41]: Thanks, that's why I added the TODO, so it could be returned to later but not now because it's not urgent.

File: docs/themes/kube/assets/dist/css/kube.css [hunk 10]
```diff
@@ -17,7 +17,7 @@ html {
 *:after {
   box-sizing: inherit; }
 
-* {
+*:not([pfelement]) {
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply docs/themes/kube/assets/dist/css/kube.css:20]: Darn you not selectors.

**Review by starryeyez024 — APPROVED** [review]
Leftovers Gone??? Thanksgiving Mayhem
